ampersand-collection.js

```diff
@@ -276,16 +276,30 @@ extend(Collection.prototype, BackboneEvents, {
         }
     },
 
-    _deIndex: function (model) {
-        for (var name in this._indexes) {
-            delete this._indexes[name][model[name] || (model.get && model.get(name))];
+    _deIndex: function (model, attribute, value) {
+        if (undefined !== attribute) {
+            if (undefined === this._indexes[attribute]) throw new Error('Given attribute is not an index');
+            delete this._indexes[attribute][value];
+            return;
+        }
+        // Not a specific attribute
+        for (attribute in this._indexes) {
+            delete this._indexes[attribute][model[attribute] || (model.get && model.get(attribute))];
         }
     },
 
-    _index: function (model) {
-        for (var name in this._indexes) {
-            var indexVal = model[name] || (model.get && model.get(name));
-            if (indexVal) this._indexes[name][indexVal] = model;
+    _index: function (model, attribute) {
+        var indexVal;
+        if (undefined !== attribute) {
```

Comment:
There's no style guide that directs comparator order, and we've already inherited some things (like no { } around single line `if` statments) that I'd personally not have chosen.   Leave it as-is imo :D